instrumentation/spring/spring-boot-autoconfigure-3/build.gradle.kts

```diff
@@ -13,7 +13,7 @@ otelJava {
 dependencies {
   val springBootVersion = "3.2.4"
   library("org.springframework.boot:spring-boot-starter-web:$springBootVersion")
-  compileOnly(project(":instrumentation:spring:spring-boot-autoconfigure"))
+  implementation(project(":instrumentation:spring:spring-boot-autoconfigure"))
```

Comment:
See https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11775#discussion_r1672309913, also applicable for `implementation`.

Reply:
Discussed, this is internal (not external) spring dep